hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -1472,55 +1479,98 @@
+            - Read the entropy_data register to fetch the 384-bit unconditioned entropy. Verify that
+              it matches with the expected value computed from the initially set LFSR.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: chip_entropy_src_cs_aes_halt
-      desc: '''Verify the aes halt handshake with CSRNG.
+      name: chip_entropy_src_fuse_en_fw_write
+      desc: '''Verify the FW override fuse input entropy_src.
 
-            Details TBD.
+            - Initialize the OTP with the entropy_src FW override fuse bit enabled.
+            - Program and enable the entropy_src in LFSR mode; program a known seed, enable the
+              entropy valid interrupt.
+            - Route the entropy data to register and disable the conditioning logic.
+            - Program the fw_ov_control to enable SW to inject entropy into the block.
+            - Write the fw_ov_wr_data register to inject the entropy data.
+            - On reception of the interrupt, read the entropy_data register - it should match what
+              was written to the fw_ov_wr_data register.
+
+            - Reset the chip and now, initialize the OTP with the entropy_src FW override fuse bit
+              disabled.
+            - Repeat the same steps as before, but this time verify that the read from the
+              entropy_data register does not match the fw_ov_wr_data, but it matches with the
+              expected value computed from the initially set LFSR.
             '''
       milestone: V2
       tests: []
     }
+
+    // CSRNG tests:
     {
-      name: chip_entropy_src_fuse_en_fw_read
-      desc: '''Verify the fuse input entropy_src.
+      name: chip_csrng_entropy_src_req
+      desc: '''Verify the CSRNG can request entropy from entropy_src.
 
-            - Initialize the OTP with this fuse bit set to 1.
-            - Perform an entropy request operation.
-            - Read the entropy_data_fifo via SW; verify that it reads valid values.
-            - Reset the chip, but this time, initialize the OTP with this fuse bit set to 0.
-            - Perform an entropy request operation.
-            - Read the internal state via SW; verify that it reads all zeros this time.
+            - Enable entropy_src in PTRNG mode.
+            - Issue an instantiate command from CSRNG to request entropy from entropy_src using the
+              SW application interface (registers).
+            - Enable cmd_req_done and entropy_req interrupts.
+            - Verify the reception of entropy_req interrupt - this should cause entropy data to flow
+              from entropy_src into csrng.
+            - Verify the reception of the cmd_done interrupt, verify that the command completed
+              without any errors.
+            - Read the genbits register to fetch the 128-bit data, and verify that it is not all 0s.
+            - We don't verify the correctness of the genbits (already done at IP level DV) - we only
```

Comment:
Reading the genbits reg here does not make sense if a gen cmd has never been issued (did not see it in the above text). What could be read is the internal state, which will be non-zero if the request from ES was done properly.